monkey/monkey_island/cc/ui/src/components/ui-components/credential-pairs/NewCredentialPair.js

```diff
@@ -87,24 +73,34 @@ const NewCredentialPair = (props) => {
   };
 
   return (
-    <XDataGrid
-      columns={getDataColumns(getRowActions, true, setErrorForRow)}
-      rows={rows}
-      rowHeight={'25px'}
-      showToolbar={false}
-      maxHeight={'400px'}
-      columnWidth={{min: 160, max: -1}}
-      hideHeaders={false}
-      processRowUpdate={processRowUpdate}
-      onRowModesModelChange={handleRowModesModelChange}
-      onRowEditStop={handleRowEditStop}
-      editMode="row"
-      rowModesModel={rowModesModel}
-      onProcessRowUpdateError={()=>{void 0;}}
-      getRowClassName={() => X_DATA_GRID_CLASSES.HIDDEN_LAST_EMPTY_CELL}
-      className={'add-new-credentials-row'}
-      setFlex={false}
-    />
+    <div>
+      <XDataGrid
+        columns={getDataColumns(null, true, setErrorForRow)}
+        rows={rows}
+        rowHeight={'25px'}
+        showToolbar={false}
+        maxHeight={'400px'}
+        columnWidth={{min: 198, max: -1}}
+        hideHeaders={false}
+        processRowUpdate={processRowUpdate}
+        onRowModesModelChange={handleRowModesModelChange}
+        onRowEditStop={handleRowEditStop}
+        editMode="row"
+        rowModesModel={rowModesModel}
+        onProcessRowUpdateError={()=>{void 0;}}
+        getRowClassName={() => X_DATA_GRID_CLASSES.HIDDEN_LAST_EMPTY_CELL}
+        className={'add-new-credentials-row'}
+        setFlex={false}
+      />
+      <Button
+        variant="contained"
+        key={nanoid()}
+        startIcon={<AddCircleIcon />}
+        disabled={errors.includes(rows?.[0]?.id)}
+        onClick={handleAddRowClick(rows?.[0]?.id)}>
+        Save
+      </Button>
```

Comment:
I don't know if we can somehow mitigate the confusion this "intermediary save" creates without reworking the table to be a table of inputs. That's the actual solution and better UX. 
If it's named "Save" we have a problem in that it doesn't actually save, you still need to "submit" the whole configuration to save it. If we name it "Add" it gives the impression that it will add a new input row. I vote to keep "Save" because it will tempt the user to press it and after it's pressed it's more clear what it does.
If we name it "Add", people with only one credential will never press it, thinking that submit is enough to save it.

Reply:
Makes sense. @ilija-lazoroski, let's go with "save" but change the icon.